haskell-src/Concordium/Types/Queries.hs

```diff
@@ -731,3 +749,218 @@ data EChainParametersAndKeys = forall (cpv :: ChainParametersVersion).
     { ecpParams :: !(ChainParameters' cpv),
       ecpKeys :: !(U.UpdateKeysCollection (AuthorizationsVersionFor cpv))
     }
+
+instance ToJSON EChainParametersAndKeys where
+    toJSON (EChainParametersAndKeys (params :: ChainParameters' cpv) keys) =
+        case chainParametersVersion @cpv of
+            SChainParametersV0 ->
+                object
+                    [ "version" .= String "0",
```

Comment:
Why is version a string? Either use a proper tag, which is normally a string, or an integer if you want to use an integer.

Reply:
THe normal way we would do this usually is to add a `ToJSON` instance for `ChainParametersVersion` like we have it for `ProtocolVersion`. Then use that here instead of manually setting versions here.